### DISCUSSION TITLE
dereferencing pointer to incomplete type ‘X509 {aka struct x509_st}’ in OpenSSL

### DISCUSSION THREAD

**Discussion by chaitu-venky** [openssl/openssl#23783] [Q&A] [answered]
Hello, Am migrating from OpenSSL 1.0.2 to 3.0.8 am not able to access the **x509->name** property. I found functions/methods to get access to the properties like onAfter, NotBefore etc. But not for the (char *NAME). Can someone help me to fix it Thanks.

**Comment by nhorman** [comment, marked as answer]
You can use X509_get_subject_name to get the X509_NAME structure and translate that into a string with X509_NAME_online
